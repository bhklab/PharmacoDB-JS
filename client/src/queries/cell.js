import { gql } from 'apollo-boost';

/**
 * @returns - Query returns the list of cell lines with it's id and name,
 * and also returns the tissue object for the particular cell line.
 */
const getCellLinesQuery = gql`
  query getAllCellLines {
    cell_lines(all: true) {
      id
      name
      tissue {
        id
        name
      }
    }
  }
`;

/**
 * @param { Number } cellId - cell id of the cell line to be queried.
 * @returns - all the information returns by the cell lines query
 * and synonym for the cell line in different datasets.
 */
const getCellLineQuery = gql`
  query getSingleCellLine($cellId: Int!) {
    cell_line(cellId: $cellId) {
      id
      name
      tissue {
        id
        name
      }
      synonyms {
        name
        source
      }
    }
  }
`;

export {
  getCellLinesQuery,
  getCellLineQuery,
};
