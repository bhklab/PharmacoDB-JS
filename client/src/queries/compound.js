import { gql } from 'apollo-boost';


/**
 * @returns - ids and names of all the compounds
 */
const getCompoundsIdNameQuery = gql`
  query getAllCompoundsIdName {
    compounds(all: true) {
      id
      name
    }
  }
`;

/**
 * @param {boolean} all - takes a boolean value if to search all the compounds or not.
 * @returns - Query returns the list of compounds with the id and name of the compound
 * and returning the annotation object for each of the compound in the database.
 */
const getCompoundsQuery = gql`
  query getAllCompounds {
    compounds(all: true) {
      id
      name
      annotation {
        smiles
        inchikey
        pubchem
        chembl
        fda_status
      }
    }
  }
`;

/**
 * @param { number } compoundId - takes the compound id as the argument to the query.
 * @returns - Query returns info of compound object for the given id and
 * also returns all the synonyms for that compound in different datasets
 * and also providing the target object.
 */
const getCompoundQuery = gql`
  query getSingleCompound($compoundId: Int, $compoundName: String) {
    singleCompound: compound(compoundId: $compoundId, compoundName: $compoundName) {
      compound {
        id
        name
        annotation {
          smiles
          inchikey
          pubchem
          fda_status
          chembl
        }
      }
      synonyms {
        name
        source {
          id
          name
        }
      }
      targets {
        id
        name
      }
    }
  }
`;

const disableDrug = gql`
  {
    drug @client
  }
`;

export {
  getCompoundsQuery,
  getCompoundQuery,
  disableDrug,
  getCompoundsIdNameQuery,
};
