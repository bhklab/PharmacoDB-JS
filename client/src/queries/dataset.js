import { gql } from 'apollo-boost';

/**
 * @returns - Query returns the list of datasets with information about how many
 * cell lines, tissues, experiments and compounds are in those datsets.
 */
const getDatasetCountsQuery = gql`
  query getDatasetData {
    datasets {
      id
      name
      cell_count
      experiment_count
      compound_tested_count
      tissue_tested_count
    }
  }
`;

/**
 * @returns - Query returns the list of datasets with the id and name of the dataset.
 */
const getDatasetsQuery = gql`
  query getAllDatasets {
    datasets {
      id
      name
    }
  }
`;

/**
 * @param { Number } datasetId - dataset id of the dataset to be queried.
 * @returns - all the information returns by the dataset query,
 * id, and name for the dataset.
 */
const getDatasetQuery = gql`
  query getSingleDataset($datasetId: Int!, $datasetName: String!) {
    dataset(datasetId: $datasetId) {
      id
      name
    }
  }
`;

/**
 * @returns - The query returns an Array of object with the dataset object, count, type
 * and an array of objects containing the id and name of the cell lines belonging to that dataset.
 */
const getCellLinesGroupedByDatasetQuery = gql`
  query getCellLinesGroupedByDataset {
    cell_lines_grouped_by_dataset {
      dataset {
        id
        name
      }
      count
      type
      list {
        id
        name
      }
    }
  }
`;

const getDatasetCellLinesQuery = gql`
  query getDatasetCellLinesQuery($datasetId: Int!) {
    dataset(datasetId: $datasetId) {
      id,
      name,
      cells_tested {
        id
        name
      }
    }
  }
`;

const getDatasetCompoundQuery = gql`
  query getDatasetCompoundQuery($datasetId: Int!) {
    dataset(datasetId: $datasetId) {
      id,
      name,
      compounds_tested {
        id
        name
      }
    }
  }
`;

const getCountTypePerDatasetQuery = gql`
  query getCountTypePerDatasetQuery ($type: String!){
    typeCountGroupByDataset (type: $type) {
      dataset {
        id
        name
      },
      count
    }
  }
`;

export {
  getDatasetCountsQuery,
  getDatasetsQuery,
  getDatasetQuery,
  getCellLinesGroupedByDatasetQuery,
  getDatasetCellLinesQuery,
  getDatasetCompoundQuery,
  getCountTypePerDatasetQuery
};
