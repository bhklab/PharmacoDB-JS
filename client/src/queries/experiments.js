/* eslint-disable import/prefer-default-export */
import { gql } from 'apollo-boost';

/**
 * @param { Number } compoundId - takes the compound id as the argument to the query.
 * @returns - Query returns all experiments for the given compound.
 */
const getSingleCompoundExperimentsQuery = gql`
  query getSingleCompoundExperiments($compoundId: Int!) {
    experiments(compoundId: $compoundId) {
      id
      cell_line {
        id
        name
        tissue {
          id
          name
        }
      }
      tissue {
        id
        name
      }
      dataset {
        id
        name
      }
      profile {
        AAC
        IC50
      }
    }
  }
`;

export {
  getSingleCompoundExperimentsQuery,
};
