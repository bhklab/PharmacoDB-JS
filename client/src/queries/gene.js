import { gql } from 'apollo-boost';

// Fragment containing the gene id, name and annotation.
const GENE_FIELDS = gql`
  fragment GeneFields on Gene {
    id
    name
    annotation {
        gene_id
        symbol
        gene_seq_start
        gene_seq_end
        chr
        strand
    }
  }
`;

/**
 * @param {boolean} all - takes a boolean value if to search all the genes or not.
 * @returns - Query returns the list of genes with the id and name of the gene
 * and also returning the annotation object for each of the gene in the database.
 */
const getGenesQuery = gql`
    ${GENE_FIELDS}
    query getAllGenes {
        genes(all: true) {
            ...GeneFields
        }
    }
`;

/**
 * @param {number} geneId - gene id for which the data is needed.
 * @returns - the information for the queried gene.
 */
const getGeneQuery = gql`
    ${GENE_FIELDS}
    query getSingleGene($geneId: Int, $geneName: String) {
        gene(geneId: $geneId, geneName: $geneName) {
            ...GeneFields
        }
    }
`;

const getCompoundTargetsQuery = gql`
    query getCompoundTargets {
        compound_targets {
            compound_id
            targets {
                id
            }
        }
    }
`;

export { 
    getGenesQuery, 
    getGeneQuery, 
    getCompoundTargetsQuery 
};
