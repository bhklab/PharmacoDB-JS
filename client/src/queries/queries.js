import { gql } from 'apollo-boost';

/**
 * Query returns the list of compounds with the id and name.
 */
const getCompoundsQuery = gql`
    {
        compounds {
            id
            name
        }
    }
`;

/**
 * Query returns info of compound given the id.
 */
const getCompoundQuery = gql`
 query Compound($compoundId: Int!) {
    compound(compoundId: $compoundId) {
      id
      name
    }
  }
`;

const disableDrug = gql`
  {
    drug @client
  }
`;

export {
  getCompoundsQuery,
  getCompoundQuery,
  disableDrug,
};
