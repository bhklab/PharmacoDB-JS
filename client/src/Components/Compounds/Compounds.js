import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import Table from '../Table/Table';
import Layout from '../Utils/Layout';
import { getCompoundsQuery } from '../../queries/queries';
import StyledWrapper from '../../styles/utils';
import BarPlot from '../Plots/BarPlot';

const table_columns = [
  {
    Header: 'Name',
    accessor: 'name',
  },
  {
    Header: 'SMILES',
    accessor: 'smiles',
  },
  {
    Header: 'InChIKeys',
    accessor: 'inchikey',
  },
  {
    Header: 'PubChem',
    accessor: 'pubchem',
  },
  {
    Header: 'FDA Status',
    accessor: 'fda_status',
  },
];

/**
 *
 * @param {Array} data - the compound data from the compounds API.
 */
const getTableData = (data) => {
  let table_data = [];
  if (data) {
    table_data = data.compounds.map((value) => {
      const { name, annotation } = value;
      return {
        name,
        smiles: annotation.smiles,
        inchikey: annotation.inchikey,
        pubchem: annotation.pubchem,
        fda_status: annotation.fda_status,
      };
    });
  }
  return table_data;
};

/**
 *
 * @param {Boolean} loading
 * @param {Boolean} error
 * @param {Array} columns
 * @param {Array} data
 *
 * @returns - (
 *  <h3> Title for barplot </h3>
 *  <BarPlot/>
 *  <h2> Title for the table </h2>
 *  <Table/>
 * )
 */
const renderComponent = (loading, error, columns, data) => {
  if (loading) {
    return <p> Loading... </p>;
  }
  if (error) {
    return <p> Error! </p>;
  }
  return (
    <>
      <h3> Number of Compounds Tested in Each Dataset </h3>
      <BarPlot />
      <h2> List of Compounds </h2>
      <Table columns={columns} data={data} />
    </>
  );
};

/**
 * Parent component for the compounds page.
 *
 * @component
 * @example
 *
 * return (
 *   <Compounds/>
 * )
 */
const Compounds = () => {
  const { loading, error, data: compound_data } = useQuery(getCompoundsQuery);
  const columns = React.useMemo(() => table_columns, []);
  const data = React.useMemo(() => getTableData(compound_data));
  return (
    <Layout>
      <StyledWrapper>
        {
          renderComponent(loading, error, columns, data)
        }
      </StyledWrapper>
    </Layout>
  );
};

export default Compounds;
