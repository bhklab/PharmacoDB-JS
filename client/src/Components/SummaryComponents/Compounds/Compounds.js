import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import Table from '../../UtilComponents/Table/Table';
import Layout from '../../UtilComponents/Layout';
import { Link } from 'react-router-dom';
import { getCompoundsQuery } from '../../../queries/compound';
import StyledWrapper from '../../../styles/utils';
import BarPlot from '../../Plots/BarPlot';
import Loading from '../../UtilComponents/Loading';

// links for pubchem and dtc.
const PUBCHEM_LINK = 'https://pubchem.ncbi.nlm.nih.gov/compound/';
const CHEMBL_LINK = 'https://www.ebi.ac.uk/chembl/compound_report_card/';

const table_columns = [
  {
    Header: 'Name',
    accessor: 'name',
    Cell: (row) => (<Link to={`/compounds/${row.row.original.uid}`}>{row.value}</Link>),
  },
  {
    Header: 'Datasets',
    accessor: 'dataset',
    Cell: (item) => {
      let datasets = item.cell.row.original.dataset;
      return (datasets.map((obj, i) => (
        obj.id
          ? (
            <span key={i}>
              <a href={`/datasets/${obj.id}`}>{obj.name}</a>{i + 1 < datasets.length ? ', ' : ''}
            </span>
          )
          : (
            <span key={i}>{obj.name}</span>
          )
        )
      ));
    }
  },
  {
    Header: 'SMILES',
    accessor: 'smiles',
    Cell: (item) => {return item.value || 'NA'} ,
  },
  {
    Header: 'InChIKeys',
    accessor: 'inchikey',
  },
  {
    Header: 'PubChem',
    accessor: 'pubchem',
    Cell: (item) => {
      let pubchem = item.cell.row.original.pubchem;
      if(pubchem){
        return(pubchem.map((id, i) => (
          <span key={i}>
            <a href={`${PUBCHEM_LINK}${id}`} target='_blank' rel='noopener noreferrer'>{id}</a>{ i + 1 < pubchem.length ? ', ' : ''}
          </span>)
        ));
      }else{
        return '';
      }
    }
  },
  {
    Header: 'ChEMBL',
    accessor: 'chembl',
    Cell: (row) => (<a href={`${CHEMBL_LINK}${row.value}`} target='_blank' rel='noopener noreferrer'>{row.value}</a>),
  },
  {
    Header: 'FDA Status',
    accessor: 'fda_status',
  },
];

/**
 *
 * @param {Array} data - compound data from the compounds API call.
 */
const getTableData = (data) => {
  let table_data = [];
  if (data) {
    table_data = data.compounds.map((value) => {
      const { name, annotation, id, uid, dataset } = value;
      const {
        smiles, inchikey, fda_status, chembl
      } = annotation;
      const pubchem = annotation.pubchem ? annotation.pubchem.split("///") : null;
      return {
        id,
        name,
        uid,
        smiles,
        inchikey,
        pubchem,
        fda_status,
        chembl,
        dataset
      };
    });
  }
  return table_data;
};

/**
 *
 * @param {Boolean} loading
 * @param {Boolean} error
 * @param {Array} columns
 * @param {Array} data
 *
 * @returns - (
 *  <h3> Title for the barplot </h3>
 *  <BarPlot/>
 *  <h2> Title for the table </h2>
 *  <Table/>
 * )
 */
const renderComponent = (loading, error, columns, data) => {
  if (loading) {
    return <Loading />;
  }
  if (error) {
    return <p> Error! </p>;
  }
  return (
    <>
      <h2 className="new-section"> Number of Compounds Tested in Each Dataset </h2>
      <BarPlot />
      <h2 className="new-section">List of Compounds</h2>
      <Table columns={columns} data={data} defaultSort={[{ id: 'fda_status' }]} />
    </>
  );
};

/**
 * Parent component for the compounds page.
 *
 * @component
 * @example
 *
 * return (
 *   <Compounds/>
 * )
 */
const Compounds = () => {
  const { loading, error, data: compound_data } = useQuery(getCompoundsQuery, {fetchPolicy: "cache-first"});
  const columns = React.useMemo(() => table_columns, []);
  const data = React.useMemo(() => getTableData(compound_data), [compound_data]);
  return (
    <Layout page="compounds">
      <StyledWrapper>
        {
          renderComponent(loading, error, columns, data)
        }
      </StyledWrapper>
    </Layout>
  );
};

export default Compounds;
