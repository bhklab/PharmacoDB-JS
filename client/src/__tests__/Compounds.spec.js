import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from '@apollo/react-testing';
import { BrowserRouter } from 'react-router-dom';
import Compounds from '../Components/Compounds/Compounds';
import { getCompoundsQuery } from '../queries/queries';

const mocks = [
  {
    request: {
      query: getCompoundsQuery,
    },
    result: {
      data: {
        compounds: { id: 1, name: 'testDrug' },
      },
    },
  },
];

it('renders Compounds correctly with a mocked query', () => {
  // must use mount to access Compounds (because child component)
  const component = mount(
    <MockedProvider mocks={mocks} addTypename={false}>
      <BrowserRouter>
        <Compounds />
      </BrowserRouter>
    </MockedProvider>,
  );
  expect(component).toMatchSnapshot();
});
