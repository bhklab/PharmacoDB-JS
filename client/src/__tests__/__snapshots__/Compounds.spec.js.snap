// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Compounds correctly with a mocked query 1`] = `Promise {}`;
